sys/riscv/riscv/pmap.c

```diff
@@ -659,6 +687,12 @@ pmap_bootstrap(vm_offset_t l1pt, vm_paddr_t kernstart, vm_size_t kernlen)
 
 	virtual_avail = roundup2(freemempos, L2_SIZE);
 	virtual_end = VM_MAX_KERNEL_ADDRESS - L2_SIZE;
+#ifdef __CHERI_PURE_CAPABILITY__
+	virtual_avail = (vm_ptr_t)cheri_setbounds((void *)virtual_avail,
```

Comment:
I remember thinking that we can clear these after vm initialization during boot, so we have two large capabilities less floating around, so probably we don't care about representability here.

Reply:
There's also not much we can do in terms of the wiggle room as virtual_end has to be at this fixed address since we use that fixed address to map the DTB and for the devmap.

I do think that virtual_avail and virtual_end end up being used post-boot though as they are the bounds that pmap_growkernel() can operate in.  Their values are also saved in the vm_map structure I think.  We could perhaps at least truncate virtual_end to a length of 0, but that doesn't seem very useful given that virtual_avail is still around.

Reply:
Yeah I have been thinking about adjusting the kernel map capability to be representable, but we would end up to drop some physical memory as well. It is also not clear what benefit we would get, since the capability for the kernel_map is owned by vm_map, which will obey the min/max settings regardless of the root capability it is given. Even from a compartmentalization standpoint I don't think it would matter since if vm_map is broken we would probably be out of luck in any case.